### DISCUSSION TITLE
Advanced Search

### DISCUSSION THREAD

**Discussion by theoldbones** [AprilSylph/XKit-Rewritten#1308] [Suggestion Box]
No clue if this is possible with Tumblr (I have no skill with coding) but as a qualitative researcher it would be really cool to have a way of fine tuning searches. Being able to search for multiple key words or phrases with Boolian operators, setting minimum reply or reblog filters, giving specific time frame to source posts from (retrieve relevant posts from specifically Oct 1, 2023 - Oct 17, 2023, for example), etc. At the moment the most fine tuning you can do is sort by post type but that still places a lot of limitations on what can be done.
Twitter has this with the advanced search on desktop but having this for Tumblr would open up so many possibilities!

**Comment by marcustyphoon** [comment]
Unfortunately, that I know of, Tumblr would have to implement these kinds of features in their back end (and expose them via their API) in order for us to have access to them!

**Reply by AprilSylph** [reply]
Yep. Or we'd have to build a mirror for all Tumblr posts, which would be monumentally stupid for several reasons.

**Reply by theoldbones** [reply]
I think tumblr has their API available, I just have no idea how I would go about using it for something like this :')

**Reply by marcustyphoon** [reply]
To clarify what we're saying here: it wouldn't be possible unless Tumblr added exactly what you wanted to said API (or made the API unusually/impractically powerful, like letting API users run whatever database commands they wanted)!

**Reply by AprilSylph** [reply]
Even with the API available it doesn't have the filtering features you want. Searching for multiple keywords, for example, is something the API doesn't support, so it can't be done. (Unless you use the API to collect posts indiscriminately and then build your own thing for picking things out, which is terribly inefficient and against Tumblr's API terms of use.)

Edit: Jinx!

**Reply by theoldbones** [reply]
Ahh ok, thats a shame. Thank you for explaining though!
